src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -192,6 +212,62 @@ public File asFile() {
     }
 
     @Override
+    public Path asPath() {
+        return fileResource.toPath();
+    }
+
+    @Override
+    public URL asUrl() {
+        return null;
```

Comment:
It's strange that `FileInputResource`s can be converted to `Path` but not `URL`, since `Path` itself is convertible to `URL` -- do you want to address this now?